### PR TITLE
[VarDumper] fix very special vars handling

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #13235 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
PR is ready. Fabbot is drunk and Travis will be fixed after subtree-plitting (for what is concerned by this)

File: src/Symfony/Component/VarDumper/Tests/CliDumperTest.php [hunk 1]
```diff
@@ -104,8 +104,182 @@ public function testGet()
...
+    }
+
+    /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function testSpecialVars56()
+    {
+        if (PHP_VERSION_ID < 50600) {
+            $this->markTestSkipped('PHP 5.6 or the symfony_debug extension is required.');
```
Comment (stof) [inline src/Symfony/Component/VarDumper/Tests/CliDumperTest.php]: should you check for the extension being loaded to avoid skipping in this case ?

**Comment by stof** [comment]
AFAICT, the PHP-CS-Fixer expects the ``&`` to be near the variable name (or with spaces on both sides), not near the assignment operator, which is why it adds the space everywhere.

for the indentation change at the end, I'm quite sure it is a bug in the PHP-CS-Fixer logic to detect closures, which is not taking into account the possibility to return by reference. I submitted https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/964

**Comment by stof** [comment]
@fabpot what is actually the Symfony coding standard for the placement of the ``&`` for references ? PSR-2 does not talk about spaces around operators at all. Given that Symfony says to put spaces around operators, the PHP-CS-Fixer is enforcing the spaces. It has no special rule for the placement of references after the assignment operator, which is why it requires a space there as well. But is the fixer right for that ?

File: src/Symfony/Component/VarDumper/Tests/CliDumperTest.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Tests/CliDumperTest.php]: No, the extension does not help here, I'm going to update the message

**Comment by keradus** [comment]
It's a `Operators should be arounded by at least one space.` rule.
There is `&=` operator (`T_AND_EQUAL`), but no `=&`. It's in fact to operators.

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**